File: ios/NewCanvasViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(NewCanvasViewManager, RCTViewManager)

RCT_EXTERN_METHOD(canvasSetStyle:(nonnull NSNumber *)node
    width:(nonnull NSNumber *)width
    height:(nonnull NSNumber *)height)

RCT_EXTERN_METHOD(contextBeginPath:(nonnull NSNumber *)node)

RCT_EXTERN_METHOD(contextClosePath:(nonnull NSNumber *)node)

RCT_EXTERN_METHOD(contextLineTo:(nonnull NSNumber *)node
    x:(nonnull NSNumber *)x
    y:(nonnull NSNumber *)y)

RCT_EXTERN_METHOD(contextMoveTo:(nonnull NSNumber *)node
    x:(nonnull NSNumber *)x
    y:(nonnull NSNumber *)y)

RCT_EXTERN_METHOD(contextStroke:(nonnull NSNumber *)node
    contextAttributes:(nonnull NSString *)contextAttributes)

RCT_EXPORT_VIEW_PROPERTY(color, NSString)


@end

File: ios/NewCanvasViewManager.swift
@objc(NewCanvasViewManager)
class NewCanvasViewManager: RCTViewManager {

  var newCanvasView: NewCanvasView!

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  override func view() -> (NewCanvasView) {
    newCanvasView = NewCanvasView()
    return newCanvasView
  }

  @objc func canvasSetStyle(_ node:NSNumber, width:NSNumber, height:NSNumber) {
    newCanvasView.canvasSetStyle(withWidth: width.intValue, withHeight: height.intValue)
  }

  @objc func contextBeginPath(_ node:NSNumber) {
    newCanvasView.contextBeginPath()
  }

  @objc func contextClosePath(_ node:NSNumber) {
    newCanvasView.contextClosePath()
  }

  @objc func contextLineTo(_ node:NSNumber, x:NSNumber, y:NSNumber) {
    newCanvasView.contextLineTo(x.floatValue, y.floatValue)
  }

  @objc func contextMoveTo(_ node:NSNumber, x:NSNumber, y:NSNumber) {
    newCanvasView.contextMoveTo(x.floatValue, y.floatValue)
  }

  @objc func contextStroke(_ node:NSNumber, contextAttributes:NSString) {
    newCanvasView.contextStroke(withAttributes: contextAttributes as String)
  }
}
